### ISSUE TITLE
Missing permission "all files access"

### ISSUE THREAD

**Issue by BluePixel4k** [nextcloud/android#14135]
### ⚠️ Before posting ⚠️

- [x] This is a **bug**, not a question or an enhancement.
- [x] I've [searched for similar issues](https://github.com/nextcloud/android/issues) and didn't find a duplicate.
- [x] I've written a clear and descriptive title for this issue, not just "Bug" or "Crash".
- [x] I agree to follow Nextcloud's [Code of Conduct](https://nextcloud.com/contribute/code-of-conduct/).

### Steps to reproduce

1. Install version 3.30.6
2. Press the upload button inside the Nextcloud app
3. Navigate into a folder with files other than pictures or videos
4. Result: there are no other files listed

### Expected behaviour

It should be possible to upload files other than pictures and videos.

### Actual behaviour

The error is caused by the missing permission "all files access". 
It was possible in previous versions.

### Android version

15

### Device brand and model

Google Pixel 8

### Stock or custom OS?

Custom (explain in "additional information")

### Nextcloud android app version

3.30.6

### Nextcloud server version

30.x.x

### Using a reverse proxy?

Yes

### Server error logs

```bash

```

### Additional information

 My phone is running GrapheneOS.

**Comment by BluePixel4k** [comment]
It's look like only the gplay release is affected.
With the fdroid, release it's possible to grand the "all files permission".
For the gplay release, the permission is not available anymore.

**Comment by AndyScherzinger** [comment]
Google forced us to remove the permission and blocked all updates to the app store since July because of this. We can't bring it back for GPlay I am afraid...

**Comment by BluePixel4k** [comment]
Oh, that's really sad.
I didn't know that, because the changelog only lists that:
_Due to new restrictions imposed by Google, the auto-upload feature will no longer be able to automatically remove uploaded files._

**Comment by cybercroft** [comment]
@AndyScherzinger so what you are saying is that Google decided to impose restrictions on the kind of data we are able to synchronize without user permission? Consequently any document files can no longer be sent to the Nextcloud server when using an android device? This indeed is concerning. I would consider this issue to be of high priority. Is there any other workaround?

**Comment by tobiasKaminsky** [comment]
The "workaraound" aka the real way Android wants it is like this
- open native Files app 
- go to your desired file
- share it to Nextcloud App
--> choose desired upload folder and upload it

And yes, Google decided this, as F-Droid is remained untouched.
There we are still allowed to use this permission.

A bit of lengthy background:
Other apps were not allowed to use this permission at all, once it was introduced in 2022.
I could convince them back then, that we need this. 
But nowadays they are more strict on it and thus we needed to remove this permission.
Thus is, why it feels now like a regression / problem in UX, while it was only an exception that they allowed it for ~2 years.

**Comment by JimmyKater** [comment]
how sad this is... 
though a big thank you for 2 years of additional functionality.

Awww... and as this issue isn't really solvable... I think it can be marked closed.

**Comment by Whisprin** [comment]
thanks for this insightful background information @tobiasKaminsky .
With this it seems that this issue mixes several related, yet not identical problems:

1. the app being unable to auto upload **files other than** pictures and videos (afaik users are not informed that this is the case and why)
2. the app being unable to auto upload **pictures and videos** (may or may not be related to this, workarounds suggest removing and adding file and folder permission or set up of a custom folder for the already existing camera folder)
3. the app being unable to **move or remove files** after they have been uploaded (currently users are informed about this "[...] , the auto-upload feature will no longer be able to automatically remove uploaded files")

For 1. auto upload of other files
If the upload for files other than pictures and videos is currently expected to be broken, then user should be a) informed that this is broken in the Google build variant and b) the user should be prompted to remove the broken custom auto upload folder pairs to make sure no wrong expectations arise going forward.
If other apps, especially the ones from Google themselves, are able to do this, it sounds like another antitrust issue and stifling competition. But I don't know if that's the case.

For 2. auto upload of pictures and videos
Would be great to understand if this is related or not, and weather auto upload of pictures and videos is currently expected to work flawlessly or not after updating the app from Google Play.

For 3. (re)moving files after auto upload
The auto upload folder setting "Original file will be..." kept/moved is grayed out and **can't be changed for picture and video folders** only. However, that setting **can be changed for other folders** by the user, e.g. Documents. Clarification if this should apply to all or only some folders would be helpful.

**Comment by drhirn** [comment]
And I would love to know, if the F-Droid version will always be working.

(when switching from g-play to f-droid I realised, an export function for settings would be nice. had to configure all upload-folders again, which are many in my case)

**Comment by GuyFromMars365** [comment]
> Google forced us to remove the permission and blocked all updates to the app store since July because of this. We can't bring it back for GPlay I am afraid...

So Google declined Permission Declaration Form you submitted? Did they explain why?  Doesn't Google Drive, One Drive, Box, Dropbox all have these permission? I assume that without this permission, Nextcloud can no longer delete ANY file on Android, including files it creates? If so, what is Goolgle's expected way of handling file sync?

**Comment by Beer17HWAM** [comment]
I removed Google play version and installed Fdroid version. It's still only uploading pictures and videos, no other files. I don't get it.

**Comment by tobiasKaminsky** [comment]
> And I would love to know, if the F-Droid version will always be working.

As long as Android itself does not remove this permission at all, then Fdroid stays "safe", as Fdroid does not restrict this.



> So Google declined Permission Declaration Form you submitted? Did they explain why?

They said that our app can use other more privacy friendly options, like their new photo/video picker.
But this is just wrong, as it is not only about accessing files, but also deleting them, which cannot be possible with their proposed way.



> Nextcloud can no longer delete ANY file on Android, including files it creates?

That is luckily not the case. Our app can delete all files that it created within its own data storage.
Our app can also read most of the files, but is (and that is the big drawback) cannot move/delete them.

**Comment by tobiasKaminsky** [comment]
You can also click "+" button and then choose "upload content from other apps" and use the Android's internal file picker.
Then all files should show up again.

**Comment by branleb** [comment]
Just encountered this issue, switched from gplay to fdroid version as recommended here.

Can't confirm switching to the fdroid version is a working solution.
There is still no way to add "All files Permission" on Android 15 (Pixel 8a) and therefore no way to use the InstantUpload Feature for files other than media (audio and video) file types. So it's not only about moving/deleting, but also about reading which seems not possible/not to be working at the moment.

Is there any way this still can be archieved?
Manual chosen upload of single files by way of the "upload content from other apps" is obviously no solution here.

I used to autoupload e.g. my entire Downloads folder (which I used to do e.g. on a G60 with Android 12) so I could download for example a PDF file on the phone (because I received it there via an App, got the link easy via a mobile first/only App etc.)  and access the file almost instantly on a bigger screen device.

Another usecase would be backing up WhatsApp to something that is not Google Drive. 

Both usecases seem now impossible.

**Comment by drhirn** [comment]
> There is still no way to add "All files Permission" on Android 15 (Pixel 8a) and therefore no way to use the InstantUpload Feature for files other than media (audio and video) file types. So it's not only about moving/deleting, but also about reading which seems not possible/not to be working at the moment.

Can't confirm this. Android 15 on Pixel 8. Nextcloud app is able to upload anything and delete files. Working as expected.
Uninstalled the G-Play Version and installed the one from F-Droid. Started the app and it asked me for permissions. Configured auto-upload-folders and everything was working like before.

**Comment by BluePixel4k** [comment]
> There is still no way to add "All files Permission" on Android 15 (Pixel 8a) and therefore no way to use the InstantUpload Feature for files other than media (audio and video) file types.

For me it's working fine with the fdroid version.
You can grand the permission in the Android settings:
Settings -> Apps -> Special app access -> All files access -> Nextcloud

**Comment by branleb** [comment]
> > There is still no way to add "All files Permission" on Android 15 (Pixel 8a) and therefore no way to use the InstantUpload Feature for files other than media (audio and video) file types.
> 
> For me, it's working fine with the fdroid version. You can grant the permission in the Android settings: Settings -> Apps -> Special app access -> All files access -> Nextcloud

Thanks for the explicit instruction. Now I found and enabled the permission with the fdroid version and I can confirm, works for me too.

Previously I had looked in the wrong place. I had found this comment in issue #2036  https://github.com/nextcloud/android/issues/2036#issuecomment-2361799154 and hat misinterpreted the » Settings > **Apps > "..." >** Special access > All files access.« as  »Settings -> **Apps ->Nextcloud ->** (Permissions ->) Special Access -> All files access« and because I didnt find any such option in the Setting -> App -> Nextcloud menu or the Permissions submenu, I concluded it was still not available while I just was looking in the wrong place. My Bad.

**Comment by Wikinaut** [comment]
Same issue on all of the phones here:

Samsung A54, Android 14

**Comment by programmer-61** [comment]
I have noticed this issue because my photos auto uploaded were missing the GPS information. After switching to the F-Droid Version now the GPS information ist back in the auto uploaded photos.
So I would guess that the new way to acces the files which Google offers removes GPS information from the photos....

**Comment by Beer17HWAM** [comment]
> I have noticed this issue because my photos auto uploaded were missing the GPS information. After switching to the F-Droid Version now the GPS information ist back in the auto uploaded photos.
> So I would guess that the new way to acces the files which Google offers removes GPS information from the photos....

Exactly the same here. Running F-Droid version now and happy again. Hope it stays that way.

**Comment by BluePixel4k** [comment]
> I have noticed this issue because my photos auto uploaded were missing the GPS information.

Oh, yes, the GPS information missing, if you only grand the photo and video permission. The bug is also present in the 3.30.6 fdroid version and 3.29.3 gplay version too. But if the permission for all files are granted, then it's working fine.

It's also mentioned in another issue:
https://github.com/nextcloud/android/issues/14199#issuecomment-2558771115

It's look like, the permission ACCESS_MEDIA_LOCATION is necessary to upload pictures with GPS info in connection with picture and video only permission.

But I think we need to open another issue for this bug.

**Comment by AndyScherzinger** [comment]
> It's look like, the permission ACCESS_MEDIA_LOCATION is necessary to upload pictures with GPS info in connection with picture and video only permission.

cc @tobiasKaminsky @alperozturk96 

> But I think we need to open another issue for this bug.

That be appreciated and helps fixing the various aspects in an isolated piece by piece manor 👍

**Comment by koying** [comment]
> For 2. auto upload of pictures and videos
> Would be great to understand if this is related or not, and whether auto upload of pictures and videos is currently expected to work flawlessly or not after updating the app from Google Play.

I had the same issue and fixed it by going to "accounts-<my nextcloud-account>" on my phone, and (re-?) enabling "Files" synchronization, which was disabled for whatever reason.
Likely due to 3.30.6 as my last auto-uploaded photos dated from end November.

**Comment by sol8712** [comment]
Just because there is no technical way to solve this issue doesn't mean there's no way to make progress towards a solution.

https://github.com/nextcloud/android/issues/13871#issuecomment-2563128153
Unless anyone can find a better solution for pushing this issue foreword, this may be the best bet at solving it.

**Comment by tobiasKaminsky** [comment]
> > It's look like, the permission ACCESS_MEDIA_LOCATION is necessary to upload pictures with GPS info in connection with picture and video only permission.
> 
> cc [@tobiasKaminsky](https://github.com/tobiasKaminsky) [@alperozturk96](https://github.com/alperozturk96)
> 
> > But I think we need to open another issue for this bug.
> 
> That be appreciated and helps fixing the various aspects in an isolated piece by piece manor 👍

https://developer.android.com/training/data-storage/shared/media#media-location-permission

We can add this. 
They say "Your app requires explicit user consent to gain access to this information." and I hope, this is getting remembered, and does not need to be done for every folder/file.

**Comment by tobiasKaminsky** [comment]
My emulator does not show any GPS data in images.
Thus I created a draft PR, with an APK, that you can install in parallel: https://github.com/nextcloud/android/pull/14247#issuecomment-2563431505

@BluePixel4k can you test this?
It should ask you, if needed, for a permission when clicking "+" button (will be changed later).

**Comment by BluePixel4k** [comment]
> Thus I created a draft PR, with an APK, that you can install in parallel: #14247 (comment)
> 
> @BluePixel4k can you test this?

Thanks a lot! Yes, it's working fine for me with this version 👍
But is it possible to enable the ACCESS_MEDIA_LOCATION already before the user grants the image and video permission? Because if a user doesn't grant the permission but shares a picture via his local gallery app to the Nextcloud app, the GPS information is missing.

**Comment by NicolasGoeddel** [comment]
> I have noticed this issue because my photos auto uploaded were missing the GPS information. After switching to the F-Droid Version now the GPS information ist back in the auto uploaded photos. So I would guess that the new way to acces the files which Google offers removes GPS information from the photos....

Holy shit. That's annoying. I need that information. What the heck is Google doing? Looks like I need to find a better way to make backups of my files. Maybe Syncthing can do better.

**Comment by NicolasGoeddel** [comment]
By the way: Why can I not select photos from my gallery and share it with Nextcloud for uploading? If I do so nothing happens. But when I go into Nextcloud and upload a photo from there it works properly. But it is very unhandy and confusing.

**Comment by DragonQ** [comment]
> By the way: Why can I not select photos from my gallery and share it with Nextcloud for uploading? If I do so nothing happens. But when I go into Nextcloud and upload a photo from there it works properly. But it is very unhandy and confusing.

For me I usually have to "share with Nextcloud" twice for it to wake up and actually upload stuff.

It is a shame Google makes these key features flaky and/or impossible with each new version of Android.

**Comment by sol8712** [comment]
> > I have noticed this issue because my photos auto uploaded were missing the GPS information. After switching to the F-Droid Version now the GPS information ist back in the auto uploaded photos. So I would guess that the new way to acces the files which Google offers removes GPS information from the photos....
> 
> Holy shit. That's annoying. I need that information. What the heck is Google doing? Looks like I need to find a better way to make backups of my files. Maybe Syncthing can do better.

Sadly Syncthing devs shut down the android version just days ago, but im sure a fork will pop up soon and continue. 
Google seems to actively be pushing the more privacy conscience away from their platforms on several fronts.

> > I have noticed this issue because my photos auto uploaded were missing the GPS information. After switching to the F-Droid Version now the GPS information ist back in the auto uploaded photos. So I would guess that the new way to acces the files which Google offers removes GPS information from the photos....
> 
> Holy shit. That's annoying. I need that information. What the heck is Google doing? Looks like I need to find a better way to make backups of my files. Maybe Syncthing can do better.

Sadly Syncthing devs shut down the android version just days ago, but im sure a fork will pop up soon and continue. 
Google seems to actively be pushing the more privacy conscience away from their platforms on several fronts.

**Comment by Beer17HWAM** [comment]
Why are other apps able to do the same thing? For example Foldersync pro.

https://play.google.com/store/apps/details?id=dk.tacit.android.foldersync.full

I made a connection (pair) with my Nextcloud environment and can sync all folders on my Android device without any problem. The files a copied with all the information, also GPS coordinates. With Foldersync the Nextcloud app isn't necessary anymore to sync files to Nextcloud. Foldersync can replace the whole Nextcloud sync option! 

Am I missing something?

**Comment by jonas740** [comment]
Same for me, about 16 december 2024 it stopped to autoupload to nextcloud. If this isn´t solved there is not that much use for nextcloud as an application in my meaning. I did however download the version from F-Droid and that seems to work. Using android 13 with Hyper OS.

**Comment by Beer17HWAM** [comment]
Just try Foldersync. It's a great tool and it has a dedicated Nextcloud feature. It can copy and also sync folders. And also to a lot of other environments.

https://play.google.com/store/apps/details?id=dk.tacit.android.foldersync.full

**Comment by jonas740** [comment]
> Just try Foldersync. It's a great tool and it has a dedicated Nextcloud feature. It can copy and also sync folders. And also to a lot of other environments.
> 
> https://play.google.com/store/apps/details?id=dk.tacit.android.foldersync.full
> 
> 
> 

Well i tried foldersync and I wasn't very happy with it.

Although it seems like the fdroid version is working for now.

But when the app isn't able to do what it's meant to do, then we have a serious problem that has to be solved.

If I want other services then nextcloud I would switch, but I have decided to go the route with nextcloud.

Hoping development department see this and give us a fix ASAP.

**Comment by BrianJKoopman** [comment]
> The "workaraound" aka the real way Android wants it is like this
> 
>     * open native Files app
> 
>     * go to your desired file
> 
>     * share it to Nextcloud App
>       --> choose desired upload folder and upload it

This "workaround" doesn't allow for files other than photos and videos.

My usual application is uploading audio recordings. Sharing a `.wav` file to the Nextcloud App doesn't upload. It doesn't show any error either. If I rename the file to `.mp4`, for instance, I can then find it when trying to select a file to upload within Nextcloud, and the upload works.

I'm using v3.30.6 on Android 15 with the "Photos and videos" permission enabled.

**Comment by jonas740** [comment]
> > The "workaraound" aka the real way Android wants it is like this
> > 
> >     * open native Files app
> > 
> >     * go to your desired file
> > 
> >     * share it to Nextcloud App
> >       --> choose desired upload folder and upload it
> 
> This "workaround" doesn't allow for files other than photos and videos.
> 
> My usual application is uploading audio recordings. Sharing a `.wav` file to the Nextcloud App doesn't upload. It doesn't show any error either. If I rename the file to `.mp4`, for instance, I can then find it when trying to select a file to upload within Nextcloud, and the upload works.
> 
> I'm using v3.30.6 on Android 15 with the "Photos and videos" permission enabled.

But it defeats the purpose of the app doesn't it?

The automated upload is pretty much the reason I use nextcloud.

I could upload to my sftp server manually if I wanted that.

Are developers working on this by the way?

**Comment by BrianJKoopman** [comment]
> But it defeats the purpose of the app doesn't it?
> 
> The automated upload is pretty much the reason I use nextcloud.

Being limited to only uploading photos/videos does, I agree. I haven't really used auto-uploads. I mostly access files on my instance with the app, and manually upload a couple files every once in a while.

**Comment by tobiasKaminsky** [comment]
> Why are other apps able to do the same thing? For example Foldersync pro.
> 
> https://play.google.com/store/apps/details?id=dk.tacit.android.foldersync.full
> 
> I made a connection (pair) with my Nextcloud environment and can sync all folders on my Android device without any problem. The files a copied with all the information, also GPS coordinates. With Foldersync the Nextcloud app isn't necessary anymore to sync files to Nextcloud. Foldersync can replace the whole Nextcloud sync option!
> 
> Am I missing something? I have a Pixel 8 pro with Android 15.

Foldersync still has the permission, that was not allowed anylonger for us:

» adb shell dumpsys package dk.tacit.android.foldersync.lite | grep MANAGE_EXT                                                                                                                                  
      android.permission.MANAGE_EXTERNAL_STORAGE


While our Gplay version:
» adb shell dumpsys package com.nextcloud.client | grep EXTERNAL | grep permission
-> nothing found

And Fdroid:
» adb shell dumpsys package com.nextcloud.client | grep EXTERNAL | grep permission
      android.permission.MANAGE_EXTERNAL_STORAGE

**Comment by Beer17HWAM** [comment]
> > Why are other apps able to do the same thing? For example Foldersync pro.
> > https://play.google.com/store/apps/details?id=dk.tacit.android.foldersync.full
> > I made a connection (pair) with my Nextcloud environment and can sync all folders on my Android device without any problem. The files a copied with all the information, also GPS coordinates. With Foldersync the Nextcloud app isn't necessary anymore to sync files to Nextcloud. Foldersync can replace the whole Nextcloud sync option!
> > Am I missing something? I have a Pixel 8 pro with Android 15.
> 
> Foldersync still has the permission, that was not allowed anylonger for us:
> 
> » adb shell dumpsys package dk.tacit.android.foldersync.lite | grep MANAGE_EXT android.permission.MANAGE_EXTERNAL_STORAGE
> 
> While our Gplay version: » adb shell dumpsys package com.nextcloud.client | grep EXTERNAL | grep permission -> nothing found
> 
> And Fdroid: » adb shell dumpsys package com.nextcloud.client | grep EXTERNAL | grep permission android.permission.MANAGE_EXTERNAL_STORAGE

Exactly! And why are they allowed?? I don't understand. That is why I mentioned it. It’s a great app and it’s working very good. Low battery usage, not even visible. Monitoring local folders and copying photos (or any other file) very quick after creation or on schedule (fully customizable) to Nextcloud. Much faster than Nextcloud. Pure a file copy without losing any data. So there is a lot to do if you want to compete with this app.

**Comment by markushaslinger** [comment]
> Exactly! And why are they allowed?? I don't understand. That is why I mentioned it. It’s a great app and it’s working very good. Low battery usage, not even visible. Monitoring local folders and copying photos (or any other file) very quick after creation or on schedule (fully customizable) to Nextcloud. Much faster than Nextcloud app. Pure a file copy without losing any data. So there is a lot to do if you want to compete with this app.

You are comparing a paid app with a free one and demand from the OSS developers to solve an issue that Google created. They are not competing, this is not a business.
If it's important for you solve the issue and send a PR, otherwise be happy with your paid solution until that stops working as well.

**Comment by MateuszKubuszok** [comment]
> Exactly! And why are they allowed?? I don't understand. That is why I mentioned it. It’s a great app and it’s working very good. Low battery usage, not even visible. Monitoring local folders and copying photos (or any other file) very quick after creation or on schedule (fully customizable) to Nextcloud. Much faster than Nextcloud app. Pure a file copy without losing any data. So there is a lot to do if you want to compete with this app.

They are allowed to do it (probably) for the same reason Nextcloud was allowed to do it: because it's reviewed manually by some people, and this time they got "you remove it or forget about publishing an update" response. No amount of "So there is a lot to do" matters if you are literally on the mercy of some randos gatekeeping someone's private app store. F-Droid has no such issues.

One day FolderSync might get the same treatment, and you find yourself with an app that won't receive any update for the next 5 years list like Nextcloud.

What I wanted to say is: it's unfair to word your demands in such way.

**Comment by tobiasKaminsky** [comment]
I assume that you understand that I cannot disclosure any conversation/mails with Google.
But I tried to convince them, via sending video, describing in length how (auto) upload will break, how others app are still using the same permission, that we had the permission since 2 years without any problems.

But, no luck.

**Comment by Wikinaut** [comment]
This is the end of Nextcloud.

**Comment by tobiasKaminsky** [comment]
If you have any idea, how, I am happy to hear.
We can only appeal against their decision and they always answered the same text blocks.
It was literally always 1:1 copy.
So it seems that they do not even try to listen to the arguments.

**Comment by koying** [comment]
I don't think not being able to sync non-photos/videos to Android is a threat to Nextcloud 😉
I'd be really happy to get back GPS info on photos, though...

**Comment by koying** [comment]
> You cannot even share upload pdf, odt or similar files now

You totally can share, at least with the f-droid version. I just tested it.
If I understand correctly the issue, you cannot *sync* anymore.

**Comment by Tasketo** [comment]
Also with the play store version.

You still can share files via the App.

You still can use Auto Upload (Just turn it on again). The files will retain on the android device rather being deleted automatically, if configured so in the past. That's not that cool, but can be cleaned manually from time to time

**Comment by DragonQ** [comment]
> Also with the play store version.
> 
> You still can share files via the App.
> 
> You still can use Auto Upload (Just turn it on again). The files will retain on the android device rather being deleted automatically, if configured so in the past. That's not that cool, but can be cleaned manually from time to time

But they also have to be manually downloaded or moved to be in the correct Nextcloud folder structure.

**Comment by Whisprin** [comment]
> If you have any idea, how, I am happy to hear. We can only appeal against their decision and they always answered the same text blocks. It was literally always 1:1 copy. So it seems that they do not even try to listen to the arguments.

Some things come to my mind:
## Improve user experience, avoid unexpected app behavior caused by the removed permission
- With the next app update, tell the users what exactly is not working (but has been working before):
  - Auto uploads of files other than photos and videos fail silently
  - Moving and removing of files (photos, videos and other files) doesn't work
  - Location data of uploaded pictures is removed silently
- Automatically prompt the users for the missing permissions after app upgrade
  - "Photos and videos" permission
  - "Location" permission (currently not present)
- Remove the settings from the "Auto upload" configuration dialog that are knowingly not working
- Ask the users to re-configure or remove their Auto upload folders if applicable 

## Build a stronger case to get the removed permission back
- Improve the app name and Play Store listing to make the apps intend clear
  - App name may be something like "Nextcloud Files & Sync" or "Nextcloud Storage & Backup"
  - "About this app" and features to include "store and backup all your files in your Nextcloud" or similar (currently it's about accessing files in Nextcloud only)
  - leave no doubt (to Google and to the user) that synchronizing arbitrary files in the background is a core feature of the app, submit the app for review again
- Conduct and publish a security audit of the app
- Connect with developers of other apps that have the "All files access" permission and see what's different

## Escalate further
 - Involve @karlitschek, the Nextcloud legal team and external help

## Informative
- Only the Google Play Store build is affected, the F-Droid build works just like before including auto uploads of arbitrary files
- Manual uploads of arbitrary files still work in the Google Play Store build variant
  - sharing a file from the Android file manager to Nextcloud results in the file being uploaded
  - picking a file from within Nextcloud using "Upload files" or "Upload content from other apps" results in the file being uploaded

(also see the email I sent you yesterday with the same points)

**Comment by BrianJKoopman** [comment]
> * Manual uploads of arbitrary files still work in the Google Play Store build variant   
>   * sharing a file from the Android file manager to Nextcloud results in the file being uploaded
>   * picking a file from within Nextcloud using "Upload files" or "Upload content from other apps" results in the file being uploaded

This wasn't true for me for whatever reason. I was unable to upload `.wav` files using either the share from the Android file manager or by picking the file within Nextcloud's file upload/content from other apps. I've since moved to the F-droid build and it's working again.

**Comment by n8falke** [comment]
> This wasn't true for me for whatever reason. I was unable to upload `.wav` files using either the share from the Android file manager or by picking the file within Nextcloud's file upload/content from other apps.

Same on my Pixel 6 Pro: No upload (file or from other app) / share / view / download of i.e. pdf.

**Comment by n8falke** [comment]
I had an updated version.

Browsing remote files and pressing on a pdf caused a download error. Pictures can still be viewed.

Upload causes a copy error.

Sharing: After choosing of target folder no upload and no error message.

**Comment by Whisprin** [comment]
I tried this
- Manually install Nextcloud app 3.30.**5** ([gplay release](https://github.com/nextcloud/android/releases/download/stable-3.30.5/gplay-release-30300590.apk) which still has the "all files access" permission)
- Grant "Full access" to storage
- Upload a .wav file using "share" -> it works
- Upgrade Nextcloud through Play Store from 3.30.5 to 3.30.**7** ("all files access" permission is removed)
- Upload the same .wav file and other files using "share" -> it still works
- Upload an .apk file using "Upload content from other apps" -> it works

**Comment by BrianJKoopman** [comment]
> Did you upgrade from an older app version? And are you testing with the latest app version installed through Google Play? [@Wikinaut](https://github.com/Wikinaut) [@BrianJKoopman](https://github.com/BrianJKoopman) [@n8falke](https://github.com/n8falke)

I was on v3.30.6, and yes upgraded from an older version, I've had the app on my phone for years at this point. Some more details in https://github.com/nextcloud/android/issues/14135#issuecomment-2571402843.

My nextcloud server version is quite out of date at the moment, so maybe that contributed. Interesting to hear `.wav` upload works for you in those scenarios.

**Comment by DragonQ** [comment]
I deleted storage then uninstalled the GPlay version, then installed the F-Droid version. Lost all my locally synced files but working fine now.

However, GPlay still insists on trying to update the app even though it wasn't installed from GPlay. How do I get it to stop trying to update it? I know I can tell it to stop _automatically_ updating it, but it still tries every time I check for updates!!

**Comment by thgoebel** [comment]
On Android 14 and above, there is the concept of "update ownership". A recent F-Droid client should request this, afaik. On older Android versions, you'll have to deal with this trying-to-update behaviour :shrug:

- https://source.android.com/docs/setup/create/app-ownership
- https://f-droid.org/2024/04/18/twif.html

**Comment by DragonQ** [comment]
> On Android 14 and above, there is the concept of "update ownership". A recent F-Droid client should request this, afaik. On older Android versions, you'll have to deal with this trying-to-update behaviour 🤷
> 
>     * https://source.android.com/docs/setup/create/app-ownership
> 
>     * https://f-droid.org/2024/04/18/twif.html

Thanks. Stuck on Android 12 until someone releases a phone I actually want so sucks for me!

**Comment by gpayer** [comment]
I got automatic photo uploading working again by doing the following steps:

* In the Nextcloud App navigate in "All files" to your normal upload folder
* Click the "+" button and select "Camera" as source. **During this step I was asked to grant media and files permissions to the app!** (this is the important part)
* Select photos and videos not uploaded yet and send it

Afterwards the automatic upload worked again. Additionally I deactivated "automatically remove permissions for unused apps" for the Nextcloud App in the app settings.

However I'm still using Android 11, so this might not work for later versions.

**Comment by Whisprin** [comment]
thanks @gpayer Could you please check if any photos that have been taken before (without the media permission) are automatically uploaded after you granted the permission?

**Comment by patvdleer** [comment]
In my case it didn't, it did start uploading again but not the missing/old one taking between mid december and the permission fix

**Comment by gpayer** [comment]
@Whisprin unfortunately that can't be tested anymore, I already uploaded all missing photos.

**Comment by silas229** [comment]
Since last night I encounter the same problem.

App version: 3.30.8 RC1
Android version: 13
Device: OnePlus Nord 2

![Image](https://github.com/user-attachments/assets/74538d0b-bf1f-4efd-ae95-70fb2bb1dfdd)

![Image](https://github.com/user-attachments/assets/6a17b16f-c691-40c1-a237-59291201a9b0)

**Comment by ados8** [comment]
Can say no solution I've tried works.
Tried the proposed older APK and FDroid but only get a single sync on setup and then any new files _will not sync._

**Comment by ados8** [comment]
Would there be any updates on this, we are getting to nearly 4 months with a critical component not functioning.

**Comment by Naeaeaeaeae** [comment]
> The android app Megasync works by letting you point to a directory that it should sync and this works with all file types. Can you consider changing the strategy for Android to how Megasync does it?

Did someone try this? 
Its a long time wirh no automated document sync …

**Comment by enoch85** [comment]
Now I'm here as well. No photos uploaded since March 2025.

OnePlus 12. Everything configured as one would expect.

**Comment by audifan88** [comment]
I'm noticing that this is really going downhill and starting to seem very inconsistent for me, sometimes it uploads, some times it doesnt, a lot of times it doesnt upload all images as it should. Everything configured, app permissions, battery optimize off, wifi only disabled etc.

**Comment by koying** [comment]
Maybe unrelated, but I found all my auto-upload folders were disabled on the app yesterday, somehow, I assume due to a recent update 🙄

**Comment by drhirn** [comment]
> all my auto-upload folders were disabled

Same here. Definitely happened after the update.

**Comment by Wikinaut** [comment]
Same here.🤦‍♂️

**Comment by programmer-61** [comment]
Same for me, I am using the fdroid version as it did not have any issue with the auto file upload. Until now at least.

**Comment by DragonQ** [comment]
> Same for me, I am using the fdroid version as it did not have any issue with the auto file upload. Until now at least.

Auto upload settings being lost in 3.31.3 is a separate issue, see here: https://github.com/nextcloud/android/issues/14931

The solution is to not upgrade to 3.31.3. If you have already then there is no solution, you just need to redo all your auto-uploads. 3.31.4, which fixes the issue, is out on F-Droid now so anyone who was purposefully skipping 3.31.3 can upgrade now.

**Comment by joho1968** [comment]
What issue(s), exactly, are solved? I'm on 3.32.1 (Google Play), and it hasn't been uploading photos for months.

**Comment by cunio** [comment]
Google Play update to 3.32.2 did not solve the problem.

**Comment by enoch85** [comment]
> Google Play update to 3.32.2 did not solve the problem.

Changed to Fdroid some time ago just for Nextcloud apps. Never looked back.

**Comment by cunio** [comment]
> > Google Play update to 3.32.2 did not solve the problem.
> 
> Changed to Fdroid some time ago just for Nextcloud apps. Never looked back.

I have been trying F-Droid version as well. NextCloud App was able to autoupload photos and videos created after its installation, nothing older.

**Comment by koying** [comment]
> NextCloud App was able to autoupload photos and videos created after its installation, nothing older.

There is a setting in the app (on the auto-upload folders) to upload already existing files.

**Comment by cunio** [comment]
> > NextCloud App was able to autoupload photos and videos created after its installation, nothing older.
> 
> There is a setting in the app (on the auto-upload folders) to upload already existing files.

which does nothing in current version

**Comment by koying** [comment]
It won't do anything on an already configured folder, indeed.
If you mean you missed some files before you switched to fdroid, you can "share" them to the app so that they are uploaded.

**Comment by enoch85** [comment]
@cunio Try reinstalling the app completely. Remove it, wipe data, and install clean from Fdroid. Watch the magic happen.

**Comment by RLalor** [comment]
I have the Fdroid version installed. On Hyper OS if I go to a photo in gallery and upload it to nextcloud. Literally nothing happens. If I go to the folder in nextcloud app and select the photo from there and try to upload, nothing happens. If I go to the actual file for the photo in the phone file browser and try to upload from there (as someone suggested), nothing at all happens on the screen.
